frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -413,7 +429,58 @@ function ReaderHighlight:addToMainMenu(menu_items)
     end
     table.insert(menu_items.highlight_options.sub_item_table, {
         text_func = function()
-            return T(_("Highlight opacity: %1"), G_reader_settings:readSetting("highlight_lighten_factor", 0.2))
+            local saved_color = self.view.highlight.saved_color or self._fallback_color
+            for __, v in ipairs(self.highlight_colors) do
+                if v[2] == saved_color then
+                    return T(_("Highlight color: %1"), string.lower(v[1]))
+                end
+            end
+            return T(_("Highlight color: %1"), saved_color)
```

Comment:
For what color?

Reply:
Not sure I understand the question ;).

But: all of them? This is to set the defaults, so you have access to all of them 'including gray).

Reply:
I still don't understand where you're going with that, sorry ;).

Reply:
If the color is unsupported (somehow, that would require some pretty heinous trickery I guess?), this would show the actual color name stored, but it would be rendered as gray.

Reply:
I mean, how did the unsupported color appear in `saved_color`?

Reply:
> If the color is unsupported (somehow, that would require some pretty heinous trickery I guess?), this would show the actual color name stored, but it would be rendered as gray.

I'm not reading all the context of this pull request, but I'd appreciate if all possible RBX hex colors (not just the ~10 or so official ones we allow users to pick) can still be displayed with in thier proper color, instead of grey.

Reply:
> I mean, how did the unsupported color appear in `saved_color`?

It shouldn't, unless someone manually mangled annotations, or we somehow fuck up the list of color names in the future.

> I'm not reading all the context of this pull request, but I'd appreciate if all possible RBX hex colors (not just the ~10 or so official ones we allow users to pick) can still be displayed in their proper color, instead of grey.

They technically can, but that requires you to user-patch the name = hash in blitbuffer for what you want to use.

(i.e., annotations store a color *name*, not a color *hash*; and it's up to the device to render that according to what it thinks that color should be mapped to. I guess the intent was to be able to move annotations between devices where some of them can be very very bad at rendering certain colors).

Reply:
> It shouldn't, unless someone manually mangled annotations, or we somehow fuck up the list of color names in the future ;).

Oh, or, if, I guess, someone user-patched both blitbuffer *and* readerhighlight to *add* more color names, and that was moved to another device.

Reply:
> annotations store a color name, not a color hash

Hmm, okay. I understand the idea of different hex codes for the same color depending on the device (but also wonder if that would ever actually happen).

My [eventual idea](https://github.com/stelzch/zotero.koplugin/issues/13) is syncing these annotations between KOReader and Zotero and since Zotero stores hex codes, not color names (and allows any hex combination), I imagined this would be similar. But, that's just hypothetical and may never happen so doesn't make sense to design it here just for that. Still, that's my reasoning.

Reply:
See the original discussion you were part of last year at the top of this issue, I didn't come up with that, I just picked up the baton ;).

Reply:
It's already coded like this, and I bet a plugin could somehow make it work with this code. Like e.g. from what you said

> [a plugin] user-patched both blitbuffer and readerhighlight to add more color names, and that was moved to another device where those color names are unknown

So it's fine. Zotero doesn't have underlining/invert highlight style anyway so those I'd need to some sort of mapping in KOReader regardless.